Update k8 docs

jseldess opened the pull request:
- Add secure deployment instructions for hosted GKE
- Split insecure and local instructions into separate files

Fixes #875 
Fixes #1063 
Fixes #1265

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/2147)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/6237228410d455a49f8610d4437b95b157ef7144/

jseldess commented:
@mberhault, @a-robinson, please take a look at the hosted gce instrutions for secure deployment. That's the only approach I've validated so far. You can ignore the local deployment pages for now.

Also, the secure "Test the cluster" step relies on  https://github.com/cockroachdb/cockroach/pull/19970.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/

a-robinson commented:
The secure instructions are looking great! I ignored the local deployment pages.

---

Reviewed 8 of 11 files at r1.
Review status: 8 of 11 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[_includes/orchestration/kubernetes-limitations.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybaHZyLw3ndPk1ZUCZ:-KybaHZz13AI1ozBnkuv:b-gmjsdn) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/kubernetes-limitations.md#L11)):*
> If this is not done, the first node will bootstrap a new cluster

If we want to be precise, this can only happen if the storage is lost while all of the cockroach processes are down, because if any other cockroach pods are running when the first node restarts with new storage it will automatically join to them. Perhaps it's not worth fixing this point though given that it'll be changing so soon (https://github.com/cockroachdb/cockroach/pull/19972)

---

*[_includes/orchestration/start-kubernetes.md, line 4 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybbSp-sIPEGcDIv3gP:-KybbSp-sIPEGcDIv3gQ:b-qe99f9) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/start-kubernetes.md#L4)):*
> ```Markdown
> 
> <div class="filters filters-big clearfix">
>   <button class="filter-button" data-scope="gce-hosted">Hosted GCE</button>
> ```

This is more commonly referred to as "Google Container Engine" or "GKE". If you want to keep the more explanatory "Hosted GCE" text, I'd suggest changing this to "Hosted GCE (Google Container Engine)"

---

*[_includes/orchestration/start-kubernetes.md, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybbmOcXe1WjrTDvcfr:-KybbmOcXe1WjrTDvcfs:b-5spxpr) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/start-kubernetes.md#L27)):*
> ```Markdown
>     This creates GCE instances and joins them into a single Kubernetes cluster named `cockroachdb`.
> 
>     The process can take a while, so don't move on to the next step until you see a `Creating cluster cockroachdb...done` message and details about your cluster.
> ```

I'd replace "a while" with "a few minutes", given that it's more helpful for understanding how long to expect it to take.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 45 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeIOpRMHeXSecolKk:-KybeIOpRMHeXSecolKl:bgg37yw) ([raw file](https://github.com/cockroachdb/docs/blob/7f63a8f81c4b0948d9a7f3cb8e6417a93b798cf3/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L45)):*
> ```Markdown
> [StatefulSet](http://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/) | A StatefulSet is a group of pods treated as stateful units, where each pod has distinguishable network identity and always binds back to the same persistent storage on restart. StatefulSets are a beta feature as of Kubernetes version 1.5.
> [persistent volume](http://kubernetes.io/docs/user-guide/persistent-volumes/) | A persistent volume is a piece of local storage mounted into a pod. The lifetime of a persistent volume is decoupled from the lifetime of the pod that's using it, ensuring that each CockroachDB node binds back to the same storage on restart.<br><br>When using `minikube`, persistent volumes are external temporary directories that endure until they are manually deleted or until the entire Kubernetes cluster is deleted.
> [persistent volume claim](http://kubernetes.io/docs/user-guide/persistent-volumes/#persistentvolumeclaims) | When pods are created (one per CockroachDB node), each pod will request a persistent volume claim to “claim” durable storage for its node.
> ```

I assume it's intentional that you removed the explanation of persistent volume claims?

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeSmvPISwUZ8rZGeq:-KybeSmw0iADvzJMv7W1:b-wr1gas) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L58)):*
> ```Markdown
> As each pod is created, it issues a Certificate Signing Request, or CSR, to have the node's certificate signed by the Kubernetes CA. You must manually check and approve each node's certificates, at which point the CockroachDB node is started in the pod.
> 
> 1. Wait for pod 1 to reach status `Init:1/2`:
> ```

This is correct for now, but it will change to `Init:0/1` once https://github.com/cockroachdb/cockroach/pull/19972 is in (a bunch of the other output in this guide will change as well). Sorry for not having made that change before you started on these docs!

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 127 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybf-NBFj2_Q7Pq126O:-Kybf-NBFj2_Q7Pq126P:b-z39ofv) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L127)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     $ kubectl certificate deny default.node.cockroachdb-0
> ```

Nit, but do you think it's worth including this? It's basically documenting an edge case, and including it clutters up the doc and risks people accidentally copying this command rather than the `approve` command, getting themselves into a bad state.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfKo2-gd-C3-L1q3L:-KybfKo2-gd-C3-L1q3M:b-pepaq8) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L157)):*
> ```Markdown
>     ~~~
>     NAME                                       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS    CLAIM                           REASON    AGE
>     pvc-52f51ecf-8bd5-11e6-a4f4-42010a800002   1Gi        RWO           Delete          Bound     default/datadir-cockroachdb-0             26s
> ```

This reminds me that we should do https://github.com/cockroachdb/docs/issues/1933 soon, since I think most people will want more than 1GiB of space :)

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 229 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfd0phtqJBbvgdE_M:-Kybfd0phtqJBbvgdE_N:b-9ujiow) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L229)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     $ kubectl certificate deny default.client.root
> ```

ditto for this, especially since it was already explained above

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 232 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfgf44oTAStWDaeU2:-Kybfgf44oTAStWDaeU3:bvfrxbh) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L232)):*
> ```Markdown
>     ~~~
> 
> 5. Get the name of the new pod:
> ```

My modification to https://github.com/cockroachdb/cockroach/pull/19970 removes this step

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 251 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfneZIL0UWMYUR5z7:-KybfneZIL0UWMYUR5z8:b-vjros0) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L251)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     $ kubectl exec -it cockroachdb-client-secure-3534299328-h1vp7 -- ./cockroach sql --certs-dir=/cockroach-certs --host=cockroachdb-public
> ```

s/cockroachdb-client-secure-3534299328-h1vp7/cockroachdb-client-secure

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 300 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfuHa1gft0LTbUB7W:-KybfuHa1gft0LTbUB7X:b-c7mtsc) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L300)):*
> ```Markdown
> 3. When you're done with the SQL shell, use **CTRL + D**, **CTRL + C**, or `\q` to exit the pod.
> 
> {{site.data.alerts.callout_success}}This pod will continue running indefinitely, so any time you need to reopen the built-in SQL client or run any other <a href="cockroach-commands.html"><code>cockroach</code> client commands</a>, such as <code>cockorach node</code> or <code>cockroach zone</code>, repeat step 6 using the appropriate <code>cockroach</code> command.<br>{{site.data.alerts.end}}
> ```

s/cockorach/cockroach

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 300 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfy1Yx82arahRV2kS:-Kybfy1Yx82arahRV2kT:b-nf5c6n) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L300)):*
> ```Markdown
> 3. When you're done with the SQL shell, use **CTRL + D**, **CTRL + C**, or `\q` to exit the pod.
> 
> {{site.data.alerts.callout_success}}This pod will continue running indefinitely, so any time you need to reopen the built-in SQL client or run any other <a href="cockroach-commands.html"><code>cockroach</code> client commands</a>, such as <code>cockorach node</code> or <code>cockroach zone</code>, repeat step 6 using the appropriate <code>cockroach</code> command.<br>{{site.data.alerts.end}}
> ```

Optional, but you may want to explain that the pod can be deleted via `kubectl delete pod cockroachdb-client-secure` and recreated whenever needed.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 408 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgGiZqmfUHGWP9Yxl:-KybgGiZqmfUHGWP9Yxm:b-89a4l7) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L408)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     $ kubectl certificate deny default.node.cockroachdb-3
> ```

Ditto

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 437 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgNcm3CE5CwjyISgh:-KybgNcm3CE5CwjyISgi:bl4zhhq) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L437)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     $ kubectl delete pods,statefulsets,services,persistentvolumeclaims,persistentvolumes,poddisruptionbudget,deployments \
> ```

You can remove the ",deployments" now that I've touched up https://github.com/cockroachdb/cockroach/pull/19970

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 489 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybg_Gb9km1DodTp9lB:-Kybg_Gb9km1DodTp9lC:bf9yyty) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L489)):*
> ```Markdown
>     certificatesigningrequest "default.node.cockroachdb-2" deleted
>     certificatesigningrequest "default.node.cockroachdb-3" deleted
>     certificatesigningrequest "node-csr-0Xmb4UTVAWMEnUeGbW4KX1oL4XV_LADpkwjrPtQjlZ4" deleted
> ```

You don't actually want/need to delete the node CSRs

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 528 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgkeimynJ3EP7LAGx:-KybgkeimynJ3EP7LAGy:b7tuuxk) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L528)):*
> ```Markdown
> 7. Stop Kubernetes:
> 
>     <div class="filter-content" markdown="1" data-scope="gce-hosted">
> ```

Nit, but this section seems like it should be shared somewhere if starting Kubernetes is

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 553 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgrAYB1cFbcLjJlMz:-KybgrAYB1cFbcLjJlN-:b2ibqdj) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L553)):*
> ```Markdown
>     </div>
> 
>     {{site.data.alerts.callout_danger}}If you stop Kubernetes without first deleting resources, the remote persistent volumes will still exist in your cloud project.{{site.data.alerts.end}}
> ```

If you want to be more precise, you could replace "without first deleting resources" with "without first deleting the persistent volumes". Deleting all the other resources isn't necessary if you're just going to shut down your cluster afterwards.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 555 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybd1pAaB_82HH0jFGG:-Kybd1pAaB_82HH0jFGH:b2p1ha4) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L555)):*
> ```Markdown
>     {{site.data.alerts.callout_danger}}If you stop Kubernetes without first deleting resources, the remote persistent volumes will still exist in your cloud project.{{site.data.alerts.end}}
> 
> ## See Also
> ```

Local Kubernetes instructions seem worth linking to from here (or even in the intro to this page, if you think that'd be appropriate)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-:-Kybh0bnG4Z5MDXofqa7:bxiperj)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Review status: 8 of 11 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[_includes/orchestration/start-kubernetes.md, line 4 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybbSp-sIPEGcDIv3gP:-KyqqX8p-LcnCVfZTs35:buvvtdc) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/start-kubernetes.md#L4)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

This is more commonly referred to as "Google Container Engine" or "GKE". If you want to keep the more explanatory "Hosted GCE" text, I'd suggest changing this to "Hosted GCE (Google Container Engine)"
</blockquote></details>

Today google is [renaming](https://cloudplatform.googleblog.com/2017/11/introducing-Certified-Kubernetes-and-Google-Kubernetes-Engine.html?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed:+ClPlBl+(Cloud+Platform+Blog)) google container engine to google kubernetes engine, so now GKE is clearly the right acronym to use (for the "hosted" case - the manual case is still GCE)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 8 of 11 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[_includes/orchestration/kubernetes-limitations.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybaHZyLw3ndPk1ZUCZ:-Kyqt4pG5ELbaGvWu-6c:bss4a2b) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/kubernetes-limitations.md#L11)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

> If this is not done, the first node will bootstrap a new cluster

If we want to be precise, this can only happen if the storage is lost while all of the cockroach processes are down, because if any other cockroach pods are running when the first node restarts with new storage it will automatically join to them. Perhaps it's not worth fixing this point though given that it'll be changing so soon (https://github.com/cockroachdb/cockroach/pull/19972)
</blockquote></details>

Removed.

---

*[_includes/orchestration/start-kubernetes.md, line 4 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybbSp-sIPEGcDIv3gP:-Kyqv2-xMuF3bidKAe42:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/start-kubernetes.md#L4)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Today google is [renaming](https://cloudplatform.googleblog.com/2017/11/introducing-Certified-Kubernetes-and-Google-Kubernetes-Engine.html?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed:+ClPlBl+(Cloud+Platform+Blog)) google container engine to google kubernetes engine, so now GKE is clearly the right acronym to use (for the "hosted" case - the manual case is still GCE)
</blockquote></details>

Done.

---

*[_includes/orchestration/start-kubernetes.md, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybbmOcXe1WjrTDvcfr:-KyqvB9o4Aaul1Mn12Fn:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/_includes/orchestration/start-kubernetes.md#L27)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I'd replace "a while" with "a few minutes", given that it's more helpful for understanding how long to expect it to take.
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 45 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeIOpRMHeXSecolKk:-KyqvMC3adWpxzM68iFq:bsnryji) ([raw file](https://github.com/cockroachdb/docs/blob/7f63a8f81c4b0948d9a7f3cb8e6417a93b798cf3/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L45)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I assume it's intentional that you removed the explanation of persistent volume claims?
</blockquote></details>

I can't figure out what content you're referring to. I didn't intend to remove anything like that. Can you clarify?

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeSmvPISwUZ8rZGeq:-KyrAh0njdGIv_06rLLm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L58)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

This is correct for now, but it will change to `Init:0/1` once https://github.com/cockroachdb/cockroach/pull/19972 is in (a bunch of the other output in this guide will change as well). Sorry for not having made that change before you started on these docs!
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 127 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybf-NBFj2_Q7Pq126O:-Kyqv_aFJ9Cm4ZDqJqRz:b9x8cu5) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L127)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Nit, but do you think it's worth including this? It's basically documenting an edge case, and including it clutters up the doc and risks people accidentally copying this command rather than the `approve` command, getting themselves into a bad state.
</blockquote></details>

Agreed. Removed.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 229 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfd0phtqJBbvgdE_M:-KyqvcEiNnKAoolZ0TES:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L229)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

ditto for this, especially since it was already explained above
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 232 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfgf44oTAStWDaeU2:-KyrAZFuCHhiC73IApgS:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L232)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

My modification to https://github.com/cockroachdb/cockroach/pull/19970 removes this step
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 251 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfneZIL0UWMYUR5z7:-KyrAXTTAP85zn-krgSM:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L251)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

s/cockroachdb-client-secure-3534299328-h1vp7/cockroachdb-client-secure
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 300 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfuHa1gft0LTbUB7W:-KyqvjzpxKcWM50zfN_b:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L300)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

s/cockorach/cockroach
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 408 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgGiZqmfUHGWP9Yxl:-KyqvlcLAHZkP_eMffB_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L408)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Ditto
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 437 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgNcm3CE5CwjyISgh:-KyqvtZ2ufwaYP9Lqa4I:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L437)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

You can remove the ",deployments" now that I've touched up https://github.com/cockroachdb/cockroach/pull/19970
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 489 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybg_Gb9km1DodTp9lB:-Kyqx_Ml-s96ZD4CAPj7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L489)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

You don't actually want/need to delete the node CSRs
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 528 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgkeimynJ3EP7LAGx:-KyqwkGEtcNilSXQyVfT:b-qu6n94) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L528)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Nit, but this section seems like it should be shared somewhere if starting Kubernetes is
</blockquote></details>

Good point. Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 553 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgrAYB1cFbcLjJlMz:-KyqxkxohVPMGznEAAfk:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c57faf16c2301b5f1eabbb0ba5a891bbaabd6f11/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L553)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

If you want to be more precise, you could replace "without first deleting resources" with "without first deleting the persistent volumes". Deleting all the other resources isn't necessary if you're just going to shut down your cluster afterwards.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/df0574112400364a0a6221de5c34045f32259dc6/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c72b4a4264801d11a52bb322d580a5b30d5c5861/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5124214044584f3990f9360bbae65a27aa335a89/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/1bca26a957adb381a56cdc20fff54cfa7159cc36/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/8d40899b3fea0eab43277abc5004d76463507723/

jseldess commented:
@a-robinson, @mberhault, this is ready for another review. For now, the secure instructions focus on hosted GKE only with a note that additional methods will be supported soon. The insecure instructions still mention manual GCE and AWS.

The other change to note is the minikube instructions moving to its own page under `Get Started > Start a Local Cluster` in the sidenav.

I'll open additional docs issues for future secure k8 doc updates (for manual gce/aws and secure minikube).

a-robinson commented:
Reviewed 11 of 11 files at r2.
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[_includes/orchestration/initialize-cluster-insecure.md, line 38 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxPfFK19ADm80c87Pg:-KyxPfFK19ADm80c87Ph:bdjtxum) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/initialize-cluster-insecure.md#L38)):*
> ```Markdown
>     ~~~
> 
> 4. Confirm that the persistent volumes and corresponding claims were created successfully for all three pods:
> ```

This part isn't really necessary, but if we're going to keep it for educational purposes then it should be before running the `cluster-init` job.

---

*[_includes/orchestration/initialize-cluster-insecure.md, line 52 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxPoEyFdkLnB1Ldubk:-KyxPoEyFdkLnB1Ldubl:b-382w2q) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/initialize-cluster-insecure.md#L52)):*
> ```Markdown
>     ~~~
> 
> {{site.data.alerts.callout_success}}The StatefulSet configuration sets all CockroachDB nodes to write to <code>stderr</code>, so if you ever need access to a pod/node's logs to troubleshoot, use <code>kubectl logs &lt;podname&gt;</code> rather than checking the log on the pod itself.{{site.data.alerts.end}}
> ```

I'd replace "on the pod itself" with "on the persistent volume"

---

*[_includes/orchestration/kubernetes-limitations.md, line 5 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxQ-yK4LELEruy78rs:-KyxQ-yK4LELEruy78rt:b-7zu0xs) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/kubernetes-limitations.md#L5)):*
> ```Markdown
> #### Kubernetes Version
> 
> Kubernetes 1.7 or higher is required. Earlier versions do not support the `maxUnavailability` field and `PodDisruptionBudget` resource type used in the CockroachDB StatefulSet configuration.
> ```

We could indicate here that running on 1.6 is possible by removing that resource, but that running at earlier versions requires more substantial changes.

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 9 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxX2SBvoDnFY-JXtEZ:-KyxX2SBvoDnFY-JXtE_:b51z0ff) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L9)):*
> ```Markdown
> Orchestration systems such as Kubernetes automate the deployment, scaling, and management of containerized applications. Combined with CockroachDB's [automated sharding](frequently-asked-questions.html#how-does-cockroachdb-scale) and [fault tolerance](frequently-asked-questions.html#how-does-cockroachdb-survive-failures), they have the potential to lower operator overhead to almost nothing.
> 
> This page shows you how to orchestrate an **insecure** multi-node CockroachDB cluster locally using the Kubernetes' [`minikube`](http://kubernetes.io/docs/getting-started-guides/minikube/) tool and beta [StatefulSet](http://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/) feature. Guidance on **secure** local orchestration are coming soon.
> ```

nit, but I don't think an apostrophe is needed on Kubernetes here

also, should the end of this line be "is coming soon"?

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 11 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxXN0brNjwK8yNnhmL:-KyxXN0brNjwK8yNnhmM:bh100h8) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L11)):*
> ```Markdown
> This page shows you how to orchestrate an **insecure** multi-node CockroachDB cluster locally using the Kubernetes' [`minikube`](http://kubernetes.io/docs/getting-started-guides/minikube/) tool and beta [StatefulSet](http://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/) feature. Guidance on **secure** local orchestration are coming soon.
> 
> {{site.data.alerts.callout_info}}To orchestrate a production deployment of CockroachDB, see <a href="orchestrate-cockroachdb-with-kubernetes.html">Orchestrate CockroachDB with Kubernetes</a>.{{site.data.alerts.end}}
> ```

If this is the message we're running with everywhere then please disregard my feedback, but I'd prefer saying "To orchestrate a *secure* deployment" rather than a "production" deployment. If I was building a startup using cockroach on kubernetes, I wouldn't bother with secure mode since individual pods often aren't directly accessible outside the cluster and I wouldn't need finer-grained user roles in the database beyond "is this person authorized to run a container in the kubernetes cluster". I'd still consider such a deployment to be "production".

For what it's worth, the warning on the "Orchestrate CockroachDB with Kubernetes (Insecure)" page reads less harshly to me because it just says "in production, we strongly recommend" without implying that insecure clusters couldn't ever be considered for production use.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes-insecure.md, line 16 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxZQoLLJwl6eIju-jf:-KyxZQoLLJwl6eIju-jg:bxem2lo) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes-insecure.md#L16)):*
> ```Markdown
> {{site.data.alerts.callout_danger}}If you plan to use CockroachDB in production, we strongly recommend using a secure cluster instead. Select <strong>Secure</strong> above for instructions.{{site.data.alerts.end}}
> 
> {{site.data.alerts.callout_info}}Running a stateful application such as CockroachDB on Kubernetes requires using some of Kubernetes' more complex features at a <a href="http://kubernetes.io/docs/api/#api-versioning">beta level</a> of support. There are easier ways to run CockroachDB on Kubernetes for testing purposes, but the method presented here is destined to become a production deployment once Kubernetes matures sufficiently.{{site.data.alerts.end}}
> ```

For what it's worth, these features have been in "beta" for 11 months now. I'd personally say they're already sufficient for production deployment. It's fine if you want to leave this in, though. I think statefulsets will be graduating from beta early next year.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 45 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeIOpRMHeXSecolKk:-KyxTb2A-pShB07ukygC:b-vngkt2) ([raw file](https://github.com/cockroachdb/docs/blob/5921804911bb413820e39ae81abf797d86fd063d/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L45)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

I can't figure out what content you're referring to. I didn't intend to remove anything like that. Can you clarify?
</blockquote></details>

The term "persistent volume claim" was defined on line 45 of this file before these changes, but now it doesn't appear to be defined anywhere.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 437 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgNcm3CE5CwjyISgh:-KyxUhkX1fuV65ZvWlqv:b-xctjha) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L437)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Done.
</blockquote></details>

We should add ",jobs" though, since we're using a `job` resource to initialize the cluster.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 191 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxUFxaAQN79CKU2a0x:-KyxUFxaAQN79CKU2a0y:b-5y9fq9) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L191)):*
> ```Markdown
>     ~~~
> 
> 5. Confirm that the persistent volumes and corresponding claims were created successfully for all three pods:
> ```

As mentioned above, if we're going to include this it should come before the initialization step.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 222 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxUVSufM-RLZWmThcw:-KyxUVSufM-RLZWmThcx:b-13xib6) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L222)):*
> ```Markdown
>     ~~~
> 
>     The pod uses the `root` client certificate created earlier during to initialize the cluster, so there's not CSR approval required.
> ```

s/not/no/?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[_includes/orchestration/initialize-cluster-insecure.md, line 38 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxPfFK19ADm80c87Pg:-Kz-jbYDMqu36SHjvAEl:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/initialize-cluster-insecure.md#L38)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

This part isn't really necessary, but if we're going to keep it for educational purposes then it should be before running the `cluster-init` job.
</blockquote></details>

Done.

---

*[_includes/orchestration/initialize-cluster-insecure.md, line 52 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxPoEyFdkLnB1Ldubk:-Kz-jo5PLzhRM1o6u94z:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/initialize-cluster-insecure.md#L52)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I'd replace "on the pod itself" with "on the persistent volume"
</blockquote></details>

Done.

---

*[_includes/orchestration/kubernetes-limitations.md, line 5 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxQ-yK4LELEruy78rs:-Kz-qZC_7RmkkRDNL57O:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/_includes/orchestration/kubernetes-limitations.md#L5)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

We could indicate here that running on 1.6 is possible by removing that resource, but that running at earlier versions requires more substantial changes.
</blockquote></details>

Done.

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 9 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxX2SBvoDnFY-JXtEZ:-Kz-t27PWYEZjRrlr0ve:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L9)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

nit, but I don't think an apostrophe is needed on Kubernetes here

also, should the end of this line be "is coming soon"?
</blockquote></details>

Done.

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 11 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxXN0brNjwK8yNnhmL:-Kz-ySB56tz0KC7o7rB1:b-o1te5j) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L11)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

If this is the message we're running with everywhere then please disregard my feedback, but I'd prefer saying "To orchestrate a *secure* deployment" rather than a "production" deployment. If I was building a startup using cockroach on kubernetes, I wouldn't bother with secure mode since individual pods often aren't directly accessible outside the cluster and I wouldn't need finer-grained user roles in the database beyond "is this person authorized to run a container in the kubernetes cluster". I'd still consider such a deployment to be "production".

For what it's worth, the warning on the "Orchestrate CockroachDB with Kubernetes (Insecure)" page reads less harshly to me because it just says "in production, we strongly recommend" without implying that insecure clusters couldn't ever be considered for production use.
</blockquote></details>

I wasn't intending to make a statement here about secure vs. insecure. I was intending to link to docs for deploying a physically distributed cluster as opposed to a local cluster. I've change the language to reflect that fact and to be in line with corresponding language on other start a local cluster pages.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes-insecure.md, line 16 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxZQoLLJwl6eIju-jf:-Kz-u6QtcrExaazLfrdV:b-lczu4) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes-insecure.md#L16)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

For what it's worth, these features have been in "beta" for 11 months now. I'd personally say they're already sufficient for production deployment. It's fine if you want to leave this in, though. I think statefulsets will be graduating from beta early next year.
</blockquote></details>

I'm all for removing!

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 45 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybeIOpRMHeXSecolKk:-Kz-wIwA7EmiTIWzKPA4:b-hclhvb) ([raw file](https://github.com/cockroachdb/docs/blob/5921804911bb413820e39ae81abf797d86fd063d/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L45)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

The term "persistent volume claim" was defined on line 45 of this file before these changes, but now it doesn't appear to be defined anywhere.
</blockquote></details>

I see now. It was only defined explicitly for `minikube`, since it's referenced already in the `persistent volume` description for the cloud version. I'll add it back in for `minikube`.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfKo2-gd-C3-L1q3L:-Kz-ys-2Uq0we3SyvNbP:byu4gu1) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L157)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

This reminds me that we should do https://github.com/cockroachdb/docs/issues/1933 soon, since I think most people will want more than 1GiB of space :)
</blockquote></details>

Do you think it's worth adding these main tutorials? Or should we create a separate doc with tweaks like these? I also wonder whether we should add details about using our example load gen configs.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 300 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybfy1Yx82arahRV2kS:-Kz-x3EV4QZ6dDbNFzwm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L300)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Optional, but you may want to explain that the pod can be deleted via `kubectl delete pod cockroachdb-client-secure` and recreated whenever needed.
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 437 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybgNcm3CE5CwjyISgh:-Kz-xK8GJSSwt-b_xx3x:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L437)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

We should add ",jobs" though, since we're using a `job` resource to initialize the cluster.
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 555 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kybd1pAaB_82HH0jFGG:-Kz-xMF5YB8-lfurZq3L:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L555)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Local Kubernetes instructions seem worth linking to from here (or even in the intro to this page, if you think that'd be appropriate)
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 191 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxUFxaAQN79CKU2a0x:-Kz-wnNObG9Rvmsy54v_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L191)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

As mentioned above, if we're going to include this it should come before the initialization step.
</blockquote></details>

Done.

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 222 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxUVSufM-RLZWmThcw:-Kz-wvakyg0mrSlbim5I:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L222)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

s/not/no/?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ac002271fc28f698e46cc941cd9db6913b1d9ff3/

a-robinson commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Reviewed 10 of 10 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 11 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KyxXN0brNjwK8yNnhmL:-Kz0Q730DAQEJLAN-0MY:b-7lqqum) ([raw file](https://github.com/cockroachdb/docs/blob/8d40899b3fea0eab43277abc5004d76463507723/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L11)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

I wasn't intending to make a statement here about secure vs. insecure. I was intending to link to docs for deploying a physically distributed cluster as opposed to a local cluster. I've change the language to reflect that fact and to be in line with corresponding language on other start a local cluster pages.
</blockquote></details>

Ah, I'm sorry. LGTM

---

*[v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md, line 9 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-Kz0PzOS-UAi-gtf1PSy:-Kz0PzOS-UAi-gtf1PSz:b-vl7gmn) ([raw file](https://github.com/cockroachdb/docs/blob/ac002271fc28f698e46cc941cd9db6913b1d9ff3/v1.1/orchestrate-a-local-cluster-with-kubernetes-insecure.md#L9)):*
> ```Markdown
> Other tutorials in this section feature the automated operations built into CockroachDB, from [data replication and sharding](demo-data-replication.html) to [high availability](demo-fault-tolerance-and-recovery.html) to [automatic rebalancing](demo-automatic-rebalancing.html). On top of this built-in automation, you can use a third-party orchestration system to simplify and automate even more of your operations, from deployment to scaling to overall cluster management.
> 
> This page walks you through a simple demonstration of how to orchestrate the deployment and management of a CockroachDB cluster using the open-source Kubernetes [`minikube`](http://kubernetes.io/docs/getting-started-guides/minikube/) tool. Starting with a few configuration files, you'll quickly create a 3-node local cluster. You'll run a load generator against the cluster and then simulating node failure, watching how Kubernetes auto-restarts without the need for any manual intervention. You'll then scale the cluster with a single command before shutting the cluster down, again with a single command.
> ```

s/simulating/simulate

---

*[v1.1/orchestrate-cockroachdb-with-kubernetes.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-KybfKo2-gd-C3-L1q3L:-Kz0QSuCqyfPkRXEeCAB:bpv1vlx) ([raw file](https://github.com/cockroachdb/docs/blob/3f350f82467e5044fdd05ffa90e00ef1cf75c3c2/v1.1/orchestrate-cockroachdb-with-kubernetes.md#L157)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Do you think it's worth adding these main tutorials? Or should we create a separate doc with tweaks like these? I also wonder whether we should add details about using our example load gen configs.
</blockquote></details>

A separate doc that we could link to from these tutorials would be good.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2147#-:-Kz0QiiXYd6aeMvqGton:b-uwkowd)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/8b1146403e04a2010d3d8c51914f4d7e96c2644b/
